src/main/java/org/zalando/nakadi/service/subscription/state/CleanupState.java

```diff
@@ -33,6 +30,9 @@ public void onEnter() {
         } finally {
             try {
                 getContext().unregisterSession();
+                getContext().getZkClient().close();
```

Comment:
I believe that in case of any problems with zk unregisterSession() will throw exception, and client won't be closed.

Reply:
my thinking was that if there is a problem in `unregisterSession` it is problem with zookeeper client, which could imply it is closed, I moved the call to another try catch